core/server.go

```diff
@@ -80,16 +81,21 @@ func (s *Server) GiveMe(ctx context.Context, request *faucetpb.GiveMeRequest) (*
 		}
 	}
 
-	// validate address format
-	acc, err := sdk.GetFromBech32(request.Address, chainConfig.AccountPrefix)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, "invalid address")
+	var address []byte
+	var err error
+	if common.IsHexAddress(request.Address) {
+		address = common.HexToAddress(request.Address).Bytes()
+	} else {
```

Comment:
Propose to add some comments, it took an extra cycle too much of thinking to understand what you were doing here

```suggestion
	// Accept both Ethereum and Cosmos addresses
	if common.IsHexAddress(request.Address) {
		address = common.HexToAddress(request.Address).Bytes()
	} else {
```

Reply:
Yep, agreed, will add.